### PR TITLE
fix: checkbox disables and clears input when checked

### PR DESCRIPTION
**Pull request by RobCattonKPMG**

### CODE REVIEW

File: site/components/form/DateSelector.tsx [hunk 1]
```diff
@@ -77,7 +77,15 @@ const DateSelector = <T extends object>({
     stateUpdater,
 }: DateSelectorProps<T>): ReactElement => {
     const [dateValue, setDateValue] = useState<Date | null>(!!disabled || !value ? null : new Date(value));
-    const [errors, setErrors] = useState<ErrorInfo[]>(initialErrors);
+    const errorsToUse = !disabled ? initialErrors : initialErrors.filter((error) => !(error.id === inputId));
```
Comment (lj-kpmg) [inline site/components/form/DateSelector.tsx:80]: I don't think we should need this filter, the hook will run on the initial render so if disabled is true the errors will be cleared down by that

File: site/pages/create-disruption.tsx [hunk 2]
```diff
@@ -264,23 +246,23 @@ const CreateDisruption = ({ inputs }: CreateDisruptionProps): ReactElement => {
...
+                            disabled={pageState.inputs["publish-no-end-date-time"] !== ""}
                             disablePast
                             inputId="publish-end-date"
                             inputName="publish-end-date"
                             stateUpdater={stateUpdater}
                         />
 
                         <TimeSelector<PageInputs>
-                            display="What is the end time?"
+                            display="End time?"
```
[the remark this answers is not in the document]
Comment (RobCattonKPMG) [reply site/pages/create-disruption.tsx:261]: removed

File: site/components/form/DateSelector.test.tsx [hunk 3]
```diff
@@ -98,11 +98,34 @@ describe("DateSelector", () => {
...
-        await userEvent.click(screen.getByLabelText("What is the start date?"));
+        await userEvent.click(screen.getByLabelText("Start date"));
         await userEvent.tab();
 
         expect(screen.getByText("Select a date")).toBeTruthy();
 
         unmount();
     });
+
+    it("should validate minLength and display error", async () => {
```
[the remark this answers is not in the document]
Comment (RobCattonKPMG) [reply site/components/form/DateSelector.test.tsx:109]: renamed

File: site/components/form/DateSelector.tsx (hunk 1 above)
Comment (RobCattonKPMG) [reply site/components/form/DateSelector.tsx:80]: Removed

File: site/components/form/TimeSelector.tsx [hunk 4]
```diff
@@ -20,7 +20,16 @@ const TimeSelector = <T extends object>({
     optional = false,
     stateUpdater,
 }: TimeSelectorProps<T>): ReactElement => {
-    const [errors, setErrors] = useState<ErrorInfo[]>(initialErrors);
+    const errorsToUse = !disabled ? initialErrors : initialErrors.filter((error) => !(error.id === inputId));
```
[the remark this answers is not in the document]
Comment (RobCattonKPMG) [reply site/components/form/TimeSelector.tsx:23]: Removed

**Comment by alainajaffer** [comment]
Page state does not go back to "" when the checkbox is clicked

**Comment by RobCattonKPMG** [comment]
> Page state does not go back to "" when the checkbox is clicked - is that intended? Screenshot 2023-03-10 at 14 01 35

**Review by alainajaffer — APPROVED** [review]

**Pull request merged**